nevergrad/optimization/optimizerlib.py

```diff
@@ -473,196 +528,6 @@ def _internal_ask(self) -> base.ArrayLike:
         return self.current_bests["optimistic"].x
 
 
-@registry.register
-class OptimisticDiscreteOnePlusOne(base.Optimizer):
-    """Close to UCB, but new arms are chosen by discrete mutations from the best.
-
-    This combines the discrete 1+1 algorithm and bandits."""
-
-    def _internal_ask(self) -> base.ArrayLike:
-        if not self._num_ask:
-            return np.zeros(self.dimension)
-        if 20 * self._num_ask <= len(self.archive) ** 3:
-            return self.current_bests["optimistic"].x
-        return mutations.discrete_mutation(self.current_bests["pessimistic"].x)
-
-
-@registry.register
-class RecombiningOptimisticNoisyDiscreteOnePlusOne(base.Optimizer):
-    """Combining the discrete 1+1, noise management a la bandit, and genetic crossovers.
-
-    Close to UCB, but new arms are chosen by discrete mutations from the current best and
-    we crossover with the best every two new arms.
-    This combines the discrete 1+1, bandits and genetic crossovers.
-    """
-
-    def _internal_ask(self) -> base.ArrayLike:
-        if not self._num_ask:
-            return np.zeros(self.dimension)
-        elif 20 * self._num_ask <= len(self.archive) ** 3:
-            return self.current_bests["optimistic"].x
-        elif self._num_ask % 2 == 0 or len(self.archive) < 3:
-            return mutations.discrete_mutation(self.current_bests["pessimistic"].x)
-        else:
-            return mutations.crossover(self.current_bests["pessimistic"].x,
-                                       mutations.get_roulette(self.archive, num=2))
-
-
-@registry.register
-class DoubleFastGADiscreteOnePlusOne(base.Optimizer):
-    """Close to discrete 1+1, but new arms are chosen by double-FastGA mutations from the current best.
-    Doerr et al, Fast Genetic Algorithms, 2017
-    """
-
-    def _internal_ask(self) -> base.ArrayLike:
-        if not self._num_ask:
-            return np.zeros(self.dimension)
-        return mutations.doubledoerr_discrete_mutation(self.current_bests["pessimistic"].x)
-
-
-@registry.register
-class FastGAOptimisticDiscreteOnePlusOne(base.Optimizer):
```

Comment:
@teytaud `FastGAOptimisticDiscreteOnePlusOne` does not seem `optimistic` since there is no noise handling, is this normal?

Reply:
that's just the consequence of manual tuning... we can switch to the same for everyone, and/or having this as a parameter.

Reply:
I suppose you were refering to the other questions?
For this one, should we change the name to `FastGADiscreteOnePlusOne` instead? (no noise handling)

Reply:
sorry this answer was supposed to be  for the 20... I check this optimism now.

Reply:
Yes I remember distinctively that there was an issue like that... we can remove "optimistic" from the name here, and have a distinct version for the noisy case (maybe that one is not yet in Nevergrad, I'll be back on this when my two hands will be back - the real FastGAOptimisticNoisyDiscreteOnePlusOne was interesting in experiments so at some point it must be in Nevergrad but no emergency).